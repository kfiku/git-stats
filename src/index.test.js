// @ts-check

/* eslint-disable fp/no-unused-expression */
/* eslint-disable fp/no-nil */

const { developersCommitsInDates } = require('./index')

jest.mock('./helpers/exec')

describe('developersCommitsInDates', () => {
  it('should work fine', async () => {
    const result = await developersCommitsInDates([''], new Date(), new Date(), () => {})
    expect(result).toMatchSnapshot()
  })
})
