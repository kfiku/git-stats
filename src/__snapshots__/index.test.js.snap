// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`developersCommitsInDates should work fine 1`] = `
Object {
  "jan@localhost": Object {
    "author": Object {
      "email": "jan@localhost",
      "name": "Jan Kowalski",
    },
    "commits": Array [
      Object {
        "authorEmail": "jan@localhost",
        "authorName": "Jan Kowalski",
        "date": "Thu Nov 23 11:25:21 2017 +0100",
        "message": "remove unuser depts",
        "stats": Object {
          "deletions": 24,
          "filesChanged": 3,
          "insertions": 1,
        },
      },
      Object {
        "authorEmail": "jan@localhost",
        "authorName": "Jan Kowalski",
        "date": "Thu Sep 28 16:25:31 2017 +0200",
        "message": "first commit",
        "stats": Object {
          "deletions": 0,
          "filesChanged": 5,
          "insertions": 280,
        },
      },
    ],
    "stats": Object {
      "deletions": 24,
      "filesChanged": 8,
      "insertions": 281,
    },
  },
  "john@localhost": Object {
    "author": Object {
      "email": "john@localhost",
      "name": "John Doe",
    },
    "commits": Array [
      Object {
        "authorEmail": "john@localhost",
        "authorName": "John Doe",
        "date": "Thu Feb 22 10:36:32 2018 +0100",
        "message": "move to lodash sortby",
        "stats": Object {
          "deletions": 13,
          "filesChanged": 3,
          "insertions": 28,
        },
      },
      Object {
        "authorEmail": "john@localhost",
        "authorName": "John Doe",
        "date": "Thu Feb 22 10:19:11 2018 +0100",
        "message": "add bin to link this cmd",
        "stats": Object {
          "deletions": 1,
          "filesChanged": 1,
          "insertions": 4,
        },
      },
      Object {
        "authorEmail": "john@localhost",
        "authorName": "John Doe",
        "date": "Thu Nov 23 11:22:42 2017 +0100",
        "message": "fix proper parsing and summit commit stats",
        "stats": Object {
          "deletions": 44,
          "filesChanged": 3,
          "insertions": 66,
        },
      },
    ],
    "stats": Object {
      "deletions": 58,
      "filesChanged": 7,
      "insertions": 98,
    },
  },
  "kfiku.com@gmail.com": Object {
    "author": Object {
      "email": "kfiku.com@gmail.com",
      "name": "Grzegorz Klimek",
    },
    "commits": Array [
      Object {
        "authorEmail": "kfiku.com@gmail.com",
        "authorName": "Grzegorz Klimek",
        "date": "Thu Feb 22 10:36:48 2018 +0100",
        "message": "cleanup",
        "stats": Object {
          "deletions": 1,
          "filesChanged": 1,
          "insertions": 0,
        },
      },
      Object {
        "authorEmail": "kfiku.com@gmail.com",
        "authorName": "Grzegorz Klimek",
        "date": "Thu Feb 22 10:19:33 2018 +0100",
        "message": "add option to pass multiple dirs to search as args",
        "stats": Object {
          "deletions": 26,
          "filesChanged": 1,
          "insertions": 44,
        },
      },
      Object {
        "authorEmail": "kfiku.com@gmail.com",
        "authorName": "Grzegorz Klimek",
        "date": "Thu Feb 22 09:55:27 2018 +0100",
        "message": "add option to use with command args",
        "stats": Object {
          "deletions": 3,
          "filesChanged": 1,
          "insertions": 3,
        },
      },
      Object {
        "authorEmail": "kfiku.com@gmail.com",
        "authorName": "Grzegorz Klimek",
        "date": "Thu Feb 22 09:55:02 2018 +0100",
        "message": "update depts",
        "stats": Object {
          "deletions": 9,
          "filesChanged": 2,
          "insertions": 9,
        },
      },
      Object {
        "authorEmail": "kfiku.com@gmail.com",
        "authorName": "Grzegorz Klimek",
        "date": "Thu Feb 22 09:08:39 2018 +0100",
        "message": "fix errors with multiline commits",
        "stats": Object {
          "deletions": 1,
          "filesChanged": 1,
          "insertions": 2,
        },
      },
      Object {
        "authorEmail": "kfiku.com@gmail.com",
        "authorName": "Grzegorz Klimek",
        "date": "Mon Oct 30 08:58:30 2017 +0100",
        "message": "add sort by lines added",
        "stats": Object {
          "deletions": 2,
          "filesChanged": 1,
          "insertions": 5,
        },
      },
      Object {
        "authorEmail": "kfiku.com@gmail.com",
        "authorName": "Grzegorz Klimek",
        "date": "Thu Sep 28 14:14:32 2017 +0200",
        "message": "Initial commit",
        "stats": Object {
          "deletions": 0,
          "filesChanged": 3,
          "insertions": 82,
        },
      },
    ],
    "stats": Object {
      "deletions": 42,
      "filesChanged": 10,
      "insertions": 145,
    },
  },
}
`;
